reporters/plain_reporter.py

```diff
@@ -12,8 +13,52 @@ def handle_message(self, msg):
 
     def print_message_ids(self):
         # Sort the messages by their type.
-        self._messages.sort(key=lambda s: s[0])
-
+        #self._messages.sort(key=lambda s: s[0])
+        self.sort_messages()
         for msg in self._messages:
             code = msg.msg_id
             print(code, '({})\n    {}'.format(msg.symbol, msg.msg))
+
+    def sort_messages(self):
+        # Sort the messages by their type.
+        self._messages.sort(key=lambda s: s[0])
+
+        i = 0
+        while i < len(self._messages):
+            current_id = self._messages[i].msg_id
+            count = 1
+            messages = ''
+            msg_count = 1
+            while i+1 < len(self._messages) and self._messages[i+1].msg_id == current_id:
+                count += 1
+                if msg_count <= 5:
+                    messages += '\n    ' + self._messages[i+1].msg
+                    msg_count += 1
+                self._messages.pop(i+1)
+
+            msg = self._messages[i].msg
+
+            if count > 1:
+                msg = 'This error occurs at '+ str(count) + ' places:\n    ' + self._messages[i].msg + messages
+                #temp = self._messages[i].__setattr__(self._messages[i].msg, 'hello')
+            #elif count == 1:
+            #    msg = self._messages[i].msg + 'This error occurs at 1 place.'
+
+            msgid = self._messages[i].msg_id
+            symbol = self._messages[i].symbol
+            confidence = self._messages[i].confidence
+            abspath = self._messages[i].abspath
+            path = self._messages[i].path
+            module = self._messages[i].module
+            obj = self._messages[i].obj
+            line = self._messages[i].line
+            column = self._messages[i].column
+            self._messages[i] = Message(msgid, symbol, (abspath, path,
```

Comment:
You can use the `_replace` method for `namedtuple` here to just replace the value of one field. Note that this method returns a new namedtuple, so you'd need to do something like `self._messages[i] = self._message[i]._replace(...)`